public/app/features/alerting/unified/hooks/useExternalAmSelector.ts

```diff
@@ -33,7 +34,8 @@ export function useExternalAmSelector(): AlertmanagerConfig[] | [] {
     } else {
       let found = false;
       for (const activeAM of discoveredAlertmanagers.activeAlertManagers) {
-        if (activeAM.url === `${url}/api/v2/alerts`) {
+        let activeUrl = new URL(activeAM.url)
+        if (activeUrl.hostname === cfgUrl.hostname && activeUrl.pathname.startsWith(cfgUrl.pathname) && activeUrl.protocol === cfgUrl.protocol) {
```

Comment:
I don't think this is enough. We need to match up in some for most of the details - including auth.

Thinking about the cloud use case, where we have 1000s of users for the same URL and path.

Reply:
It is very likely we'll have to change this upstream or do some work to adjust the matching within Grafana.